portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-download.controller.js

```diff
@@ -5,30 +5,36 @@ const { FILESHARES } = require('../../../constants/file-upload');
 
 /**
  * Fetches metadata about the utilisation report file associated with the
- * specified MongoDB _id
- * @param _id {string} - the MongoDB _id of the report
+ * specified id and bankId
+ * @param bankId {string} - the id of the bank
+ * @param id {number} - the SQL id of the report
  * @returns {Promise<{ filename: string; mimetype: string }>}
  */
-const getUtilisationReportFileMetadata = async (_id) => {
-  const { azureFileInfo } = await api.getUtilisationReportById(_id);
+const getUtilisationReportFileMetadata = async (bankId, id) => {
+  const response = await api.getUtilisationReportById(id);
+  const { azureFileInfo, bankId: reportBankId } = response;
+
+  if (reportBankId !== bankId) {
+    throw new Error(`Failed to find utilisation report with bankId ${bankId} and report id ${id}`)
+  }
```

Comment:
What was the reason for adding this check?

Reply:
So from my first read of the code I had thought it was possible to use an id for a report for a different bank and successfully download it, which would of course be a security issue. Mike pointed out that the folder name being the bank id prevents against that, but he suggested we add this check anyway. Doing it explicitly does a couple of things:
1. Makes the code more readable and future proof by making it clear that it should fail if a user from one bank tries to access another bank's reports.
2. Avoids us bothering to try to fetch a document we know won't be found and allows us to differentiate this failure from other document not found errors (usefulness probably dependent on log readability)

Reply:
It does feel a bit like there is some redundancy in checking this like you've said which was why I asked about it, but keeping this in as a form of documentation and future-proofing is fine I think :)